pkg/controller/core/localqueue_controller.go

```diff
@@ -174,11 +208,56 @@ func (h *qWorkloadHandler) Generic(e event.GenericEvent, q workqueue.RateLimitin
 	q.AddAfter(req, constants.UpdatesBatchPeriod)
 }
 
+// qCQHandler signals the controller to reconcile the Queue associated
+// to the workload in the event.
+// Since the events come from a channel Source, only the Generic handler will
+// receive events.
+type qCQHandler struct {
+	client client.Client
+	log    logr.Logger
+}
+
+func (h *qCQHandler) Create(event.CreateEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *qCQHandler) Update(event.UpdateEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *qCQHandler) Delete(event.DeleteEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *qCQHandler) Generic(e event.GenericEvent, wq workqueue.RateLimitingInterface) {
+	cq, ok := e.Object.(*kueue.ClusterQueue)
+	if !ok {
+		return
+	}
+
+	var queues kueue.LocalQueueList
+	err := h.client.List(context.Background(), &queues, client.MatchingFields{utilindexer.QueueClusterQueueKey: cq.Name})
+	if err != nil {
+		h.log.Error(err, "listing queues that match the clusterQueue")
+		return
+	} else if len(queues.Items) == 0 {
+		return
+	}
```

Comment:
Maybe you can remove these lines because we can check them by the loop below.

Reply:
It makes sense :)

Reply:
Done.